spec/turn_spec.rb

```diff
@@ -71,9 +79,44 @@
 
       columns = ["a", "b", "c", "d", "e", "f", "g"]
       board.board_grid[0][0].set_state("x")
-
+      
       expect(turn.column_is_full?("a", columns)).to be true
       expect(turn.column_is_full?("b", columns)).to be false
     end
   end
+  
+  describe "#validate_computer move" do 
+    it "validates if computers random move" do 
+      computer = Computer.new
+      board = Board.new
+      turn = Turn.new(computer, board)
+      columns = ["a", "b", "c", "d", "e", "f", "g"]
+      
+      
+      expect(turn.validate_cpu_move("b", columns)).to eq("b")
+    end
+    
+    it "returns invalid if the column is full" do 
+      computer = Computer.new
+      board = Board.new
+      turn = Turn.new(computer, board)
+      columns = ["a", "b", "c", "d", "e", "f", "g"]
+      
+      board.board_grid[0][0].set_state("x")
+      expect(turn.validate_cpu_move("a", columns)).to eq("invalid")
+    end
+  end 
+  
+  describe "#find_lowest_cell_in_column" do 
+    it "can find the lowest cell and set_player piece" do 
+      player = Player.new
+      board = Board.new
+      turn = Turn.new(player, board)
+      move = "a"
+      
+      # require 'pry';binding.pry
+      turn.find_lowest_cell_in_column(move, turn.columns)
+      expect(board.board_grid[4][0].state).to eq(".")
```

Comment:
Maybe to make this more clear we can add another expect for the grid one row beneath where it's empty and show that the piece was set on the lowest possible row as well?